pennylane/tape/interfaces/autograd.py

```diff
@@ -149,8 +149,22 @@ def get_parameters(self, trainable_only=True, return_arraybox=False):
 
         return params if return_arraybox else autograd.builtins.list(params)
 
-    @autograd.extend.primitive
     def _execute(self, params, device):
+        """Wrapper to cath the needless autograd forward pass
+
+        This function is not primitive to autograd so when params
+        is a autograd box we know that the following call to
+        the primitve wrapped function is the autograd forward pass.
+        """
+        if isinstance(params, autograd.extend.Box):
+            self.do_not_execute_device_as_this_is_the_needless_autograd_forward_pass = True
+        else:
+            self.do_not_execute_device_as_this_is_the_needless_autograd_forward_pass = False

```

Comment:
It is not super elegant to set a member variable outside of `__init__()` but as this is merely a mixin its `__init__()` is never called. Also below properties of self are accessed that are not defined in this class or any super class but then defined in practice when this mixin is used.